### ISSUE TITLE
Vehicle restrictions not having an effect?

### ISSUE THREAD

**Issue by mmc1718** [VROOM-Project/vroom#893]
I have a scenario with two vehicles and am using capacity limits. In the result some shipments are not assigned, but only one vehicle is being used. After reading advice in other issues relating to only one vehicle being used (I understand this might be the cheapest solution) I have tried reducing the vehicle use cost and adding max_time_limit to the vehicles, but these seem to have no effect. 

So I have two questions:

- Am I doing something wrong with the cost and max time limit?
- Is it expected behaviour that jobs are left out even though there is an extra unused vehicle available?

Any advice would be much appreciated! Thanks :)

My request is:

```{
    "jobs": [],
    "shipments": [
        {
            "amount": [
                125
            ],
            "pickup": {
                "id": 1,
                "description": "Eubabrunner",
                "location": [
                    12.375612,
                    50.311585
                ]
            },
            "delivery": {
                "id": 1,
                "description": "Brassertstraße",
                "location": [
                    7.006184,
                    51.435145
                ]
            }
        },
        {
            "amount": [
                75
            ],
            "pickup": {
                "id": 2,
                "description": "Eubabrunner ",
                "location": [
                    12.372836,
                    50.312032
                ]
            },
            "delivery": {
                "id": 2,
                "description": "Servatiusstraße",
                "location": [
                    7.128361,
                    50.698487
                ]
            }
        },
        {
            "amount": [
                250
            ],
            "pickup": {
                "id": 3,
                "description": "Eubabrunner ",
                "location": [
                    12.372836,
                    50.312032
                ]
            },
            "delivery": {
                "id": 3,
                "description": "Hans-Mielich-Straße",
                "location": [
                    11.5701,
                    48.114105
                ]
            }
        },
        {
            "amount": [
                125
            ],
            "pickup": {
                "id": 4,
                "description": "Eubabrunner ",
                "location": [
                    12.372836,
                    50.312032
                ]
            },
            "delivery": {
                "id": 4,
                "description": "Niesenbergergasse",
                "location": [
                    15.426338,
                    47.070427
                ]
            }
        },
        {
            "amount": [
                100
            ],
            "pickup": {
                "id": 5,
                "description": "Eubabrunner ",
                "location": [
                    12.372836,
                    50.312032
                ]
            },
            "delivery": {
                "id": 5,
                "description": "Baseler",
                "location": [
                    13.544457,
                    52.648659
                ]
            }
        },
        {
            "amount": [
                100
            ],
            "pickup": {
                "id": 6,
                "description": "Eubabrunner",
                "location": [
                    12.375612,
                    50.311585
                ]
            },
            "delivery": {
                "id": 6,
                "description": "Kranichweg",
                "location": [
                    10.007046,
                    53.576158
                ]
            }
        },
        {
            "amount": [
                15
            ],
            "pickup": {
                "id": 7,
                "description": "Eubabrunner",
                "location": [
                    12.375612,
                    50.311585
                ]
            },
            "delivery": {
                "id": 7,
                "description": "Brassertstraße",
                "location": [
                    7.006184,
                    51.435145
                ]
            }
        },
        {
            "amount": [
                100
            ],
            "pickup": {
                "id": 8,
                "description": "Eubabrunner",
                "location": [
                    12.375612,
                    50.311585
                ]
            },
            "delivery": {
                "id": 8,
                "description": "Schopenhauerstraße",
                "location": [
                    11.089087,
                    49.468578
                ]
            }
        },
        {
            "amount": [
                75
            ],
            "pickup": {
                "id": 9,
                "description": "Eubabrunner",
                "location": [
                    12.375612,
                    50.311585
                ]
            },
            "delivery": {
                "id": 9,
                "description": "Jahnstraße",
                "location": [
                    9.38677,
                    52.044315
                ]
            }
        },
        {
            "amount": [
                25
            ],
            "pickup": {
                "id": 10,
                "description": "Eubabrunner",
                "location": [
                    12.375612,
                    50.311585
                ]
            },
            "delivery": {
                "id": 10,
                "description": "Nürnberger",
                "location": [
                    13.722712,
                    51.033454
                ]
            }
        },
        {
            "amount": [
                75
            ],
            "pickup": {
                "id": 11,
                "description": "Eubabrunner ",
                "location": [
                    12.372836,
                    50.312032
                ]
            },
            "delivery": {
                "id": 11,
                "description": "Niesenbergergasse",
                "location": [
                    15.426338,
                    47.070427
                ]
            }
        },
        {
            "amount": [
                75
            ],
            "pickup": {
                "id": 12,
                "description": "Eubabrunner",
                "location": [
                    12.372836,
                    50.312032
                ]
            },
            "delivery": {
                "id": 12,
                "description": "Wipplingerstraße",
                "location": [
                    16.368704,
                    48.212712
                ]
            }
        }
    ],
    "vehicles": [
        {
            "id": 1,
            "start": [
                13.46325,
                48.575009
            ],
            "startDescription": "V9",
            "end": [
                13.46325,
                48.575009
            ],
            "endDescription": "V9",
            "capacity": [
                90
            ],
            "costs": {
                "fixed": 0,
                "per_hour": 0
            },
            "max_travel_time": 28800
        },
        {
            "id": 2,
            "start": [
                13.46325,
                48.575009
            ],
            "startDescription": "V0",
            "end": [
                13.46325,
                48.575009
            ],
            "endDescription": "V0",
            "capacity": [
                95
            ],
            "costs": {
                "fixed": 0,
                "per_hour": 0
            },
            "max_travel_time": 28800
        }
    ],
    "options": {
        "g": true
    }
}
```

**Comment by jcoupey** [comment]
Having a standalone instance does expose the matrix, but mostly it allows to work on the exact same instance. Otherwise I have no way to really reproduce your problem, not with a different routing stack.

> When I increase the capacity of the van, all the shipments are completed

This seem to hint at the capacity restriction being responsible here.

**Comment by mmc1718** [comment]
I have looked into this more - you were right, the capacity restrictions were making the unassigned tasks impossible.

I misunderstood how `max_travel_time` and now realise it means time of continuous driving, not overall route. I have switched to `time_window` and I can see both vehicles are now being used.

Thanks again for your quick response on this :)

More generally - I am not a C/C++ developer - I work mostly with Python and JS - but if you are looking for contributers around e.g. documentation or more example cases I would be happy to learn more and help out.
